mage/create-a-new-module.md

```diff
@@ -0,0 +1,116 @@
+# Developing a Memgraph module in Python
+
+### Prerequisites
+- [MemgraphDB]()
+- [MAGE]()
+
+# Developing a module
+
+In this guide, we'll create a [random walk](https://en.wikipedia.org/wiki/Random_walk#On_graphs) algorithm.
+Position yourself in the MAGE repository you cloned earlier. Specifically, go in the `python` sub-directory and create a new file called `random_walk.py`.
+Next, import `mgp`, Memgraph's internal data structure module. Among others, it contains definitions for Vertex and Node data structures.
+
+> If you are using code completion, you might be interested in having `mgp` in your python path. This will allow your code editor to access `mgp.py` and make use of code declarations and documentation. `mgp.py` is a script that comes with Memgraph. Usually, it will be installed in `/usr/lib/memgraph/python_support/mgp.py`. Now, all you have to do is add it to your `PYTHONPATH` environment variable: `export PYTHONPATH=/usr/lib/memgraph/python_support`.
+
+Here's the code for the random walk algorithm:
+
+```python
+import mgp
+import random
+
+
+@mgp.read_proc
+def get(
+    start: mgp.Vertex,
+    length: int = 10,
+) -> mgp.Record(path=mgp.Path):
+    """Generates a random path of length `length` or less starting
+    from the `start` vertex.
+
+    :param mgp.Vertex start: The starting node of the walk.
+    :param int length: The number of edges to traverse.
+    :return: Random path.
+    :rtype: mgp.Record(mgp.Path)
+    """
+    path = mgp.Path(start)
+    vertex = start
+    for _ in range(length):
+        try:
+            edge = random.choice(list(vertex.out_edges))
+            path.expand(edge)
+            vertex = edge.to_vertex
+        except IndexError:
+            break
+
+    return mgp.Record(path=path)
+```
+
+The `get_path` is decorated with the `@mgp.read_proc` decorator, which tells Memgraph it's a `read` procedure, meaning it won't change the graph. The path is created from the `start` node, and edges are appended to it iteratively.
+
+## Running
+
+Start Memgraph and MAGE, and copy the module you developed into the `/usr/lib/memgraph/query_modules`.
+
+Instructions for a local Memgraph installation:
+```
+sudo systemctl start memgraph
+cp python/random_walk.py /usr/lib/memgraph/query_modules/
+```
+
+Instructions for a docker Memgraph instance:
+```
+docker run --rm -d --name memgraph -p 7687:7687 memgraph/mage
+docker cp python/random_walk.py memgraph:/usr/lib/memgraph/query_modules/
```

Comment:
In addition to this, the other possibility is having --mount, or volume set which will automatically translate local changes to the docker container storage and make work more efficient.